### PR TITLE
python3Packages.detectron2: init at 0.2.1

### PR DESCRIPTION
**Pull request by piegamesde**
###### Motivation for this change

**Looking for maintainers!**

I packaged `detectron2` and its dependencies that were not in nixpkgs (`fvcore`, `iopath`, `yacs`) for personal use. I do not want to maintain this though. I open this pull request in the hope that somebody in need will easily find this, and ideally finish what I started.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/detectron2/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    pydot
+    tqdm
+    cloudpickle
+    fvcore
+    matplotlib
+    pycocotools
+    tensorflow-tensorboard
+  ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/detectron2/default.nix:47]: Why did you disable tests?

File: pkgs/development/python-modules/detectron2/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  checkInputs = [
+    pytest
+    pytestrunner
+  ];
+
+  meta = with lib; {
+    description = "Facebooks's next-generation platform for object detection, segmentation and other visual recognition tasks";
+    homepage = "https://github.com/facebookresearch/detectron2";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/detectron2/default.nix:58]: Missing maintainer.

File: pkgs/development/python-modules/fvcore/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  pname = "fvcore";
+  version = "0.1.5.post20210402";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0zrb1dhiqdmjs6s36a2wrk4m4hry34zdy27l7rflwzc2q093rmji";
+  };
+
+  # There's an actual, proper test failure in here. Might blow up later
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fvcore/default.nix:25]: Did you try fixing it?

File: pkgs/development/python-modules/iopath/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  version = "0.1.8";
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = "iopath";
+    rev = "v${version}";
+    sha256 = "1v79z6rja4q21na31mzs6ff63as8k6gpssi2d0wpikgzk84xva0l";
+  };
+
+  # A few tests do HTTP. One could disable them individually
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/iopath/default.nix:21]: Please do that.

File: pkgs/development/python-modules/yacs/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    owner = "rbgirshick";
+    repo = "yacs";
+    rev = "v${version}";
+    sha256 = "17ah2k8hpwgsqp1l1wzvz718a9rfj5rwa5vnhpnz94akicphbvww";
+  };
+
+  nativeBuildInputs = [
+    pyyaml
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/yacs/default.nix:21]: Whats with tests?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
All are missing pythonImportsChecks.

File: pkgs/development/python-modules/detectron2/default.nix [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  version = "0.5";
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = name;
+    rev = "v${version}";
+    sha256 = "15fvf38vaxmnf5yhnn0dk9smf9gmkbs3mp1jfpxzxh4s3r05fvgd";
+  };
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/detectron2/default.nix:34]:
```suggestion
  postPatch = ''
```

File: pkgs/development/python-modules/detectron2/default.nix [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    substituteInPlace setup.py --replace "black==21.4b2" "black>=21.4b2"
+  '';
+
+  nativeBuildInputs = [
+    which
+  ];
+
+  propagatedBuildInputs = [
+    black
+    cython
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/detectron2/default.nix:44]: I think this is nativeBuildInputs, too.

File: pkgs/development/python-modules/detectron2/default.nix [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    matplotlib
+    pycocotools
+    tensorflow-tensorboard
+    hydra-core
+  ];
+
+  doCheck = false;
+
+  checkInputs = [
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/detectron2/default.nix:62]: Can we use pytestCheckHook?

File: pkgs/development/python-modules/detectron2/default.nix (hunk 7 above)
Comment (piegamesde) [reply pkgs/development/python-modules/detectron2/default.nix:44]: Which one, exactly?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/detectron2/default.nix:44]: cython.

**Comment by happysalada** [comment]
hey, thank you for this. I want to try to maintain this.
https://github.com/NixOS/nixpkgs/pull/242969
I've started with your file and made changes related to the latest version of the setup.py.
I don't have a linux machine to test though, so I have left the PR as a draft for now.
If you have time to test and have comments, I would be happy to make some changes.

**Pull request closed without merging**